src/modals/cpu-pinning/CpuPinningDataProvider.js

```diff
@@ -42,6 +50,26 @@ const getThreads = (vm) => {
   }
 }
 
+const getCpuPinningString = (vm) => {
+  // e.g 0#0_1#1-4,^2
+  const pinnings = getPinnings(vm)
+  const pinningsStrings = []
+  pinnings.forEach((pinning) =>
+    pinningsStrings.push(`${pinning.vcpu}#${pinning.cpuSet}`)
+  )
+  return pinningsStrings.join('_')
+}
+
+const getCpuToPinnedCpuMapForVm = (vm) => {
+  const pinnings = getPinnings(vm)
+  const mapping = new Map()
+  pinnings && pinnings.forEach((pinning) => {
```

Comment:
`pinnings` should be well defined here (if not then we should use this kind of checks everywhere or fix the `getPinninggs`)

btw we could get rid of intermediate variables i.e. 
```
return getPinnings(vm)
  reduce((mapping, pinning) => { ...}, new Map())
```

Reply:
Oook, lets try that...